io.podman_desktop.PodmanDesktop.yml

```
@@ -0,0 +1,76 @@
+app-id: io.podman_desktop.PodmanDesktop
+runtime: org.freedesktop.Platform
+runtime-version: "21.08"
+sdk: org.freedesktop.Sdk
+base: org.electronjs.Electron2.BaseApp
+base-version: "21.08"
+sdk-extensions:
+  - org.freedesktop.Sdk.Extension.node16
+command: run.sh
+separate-locales: false
+finish-args:
+  - "--socket=wayland"
+  - "--socket=fallback-x11"
+  - "--share=ipc"
+  - "--device=dri"
+  - "--filesystem=home"
+  - "--filesystem=xdg-run/podman"
+  - "--share=network"
+  - "--talk-name=org.freedesktop.Notifications"
+build-options:
+  append-path: /usr/lib/sdk/node16/bin
+modules:
+  # Podman Desktop sources
+  - name: podman-desktop
+    buildsystem: simple
+    build-options:
+      env:
+        XDG_CACHE_HOME: /run/build/podman-desktop/flatpak-node/cache
+        YARN_CACHE_FOLDER: /run/build/podman-desktop/flatpak-node/yarn-cache
+    build-commands:
+      # Display tooling version
+      - node --version
+      - yarn --version
+      # configure mirror to use the one generated by generated-sources.json artifacts
+      - echo "yarn-offline-mirror $FLATPAK_BUILDER_BUILDDIR/flatpak-node/yarn-mirror" > .yarnrc
+      # install in offline mode (all steps need to use offline with flatpak)
+      - yarn --offline install
+      # compile the app
+      - MODE=production yarn --offline run build
+      # Build electron app
+      - MODE=production $(npm bin)/electron-builder build --config .electron-builder.config.js --linux --dir
+      # Bundle app and dependencies
+      - cp -a dist/linux*unpacked /app/main
+      # add icons
+      - for size in {32,64,128,256,512}; do
+          rsvg-convert -w $size -h $size -f png -o "${FLATPAK_ID}.png" "buildResources/icon.svg";
+          echo "generating ${FLATPAK_ID} png file for size ${size}";
+          install -p -Dm644 "${FLATPAK_ID}.png" -t "${FLATPAK_DEST}/share/icons/hicolor/${size}x${size}/apps/";
+        done;
+        # add scalable svg icon
+      - install -p -Dm644 "buildResources/icon.svg" -t "${FLATPAK_DEST}/share/icons/hicolor/scalable/apps/${FLATPAK_ID}.svg";
+      # add metainfo
+      - install -Dm644 .flatpak-appdata.xml "${FLATPAK_DEST}/share/metainfo/${FLATPAK_ID}.metainfo.xml";
+      # Add version
+      - sed -e 's|</component>|<releases><release version="0.0.6" date="2022-08-10"></release></releases></component>|' -i "${FLATPAK_DEST}/share/metainfo/${FLATPAK_ID}.metainfo.xml";
```

It's up to you, but I'd rather not hardcode the version - it's easy to forget to update it. Instead, you can use a git-type source and extract version from tag (and date from tagged commit).
Something like this can probably work:
```suggestion
      - |
        set -e
        VERSION=$(git describe --tags --always | sed 's/^[vV]//')
        DATE=$(date -d @$(git show -s --format=%ct) +'%Y-%m-%d')
        sed -e 's|</component>|<releases><release version="'$VERSION'" date="'$DATE'"></release></releases></component>|' -i "${FLATPAK_DEST}/share/metainfo/${FLATPAK_ID}.metainfo.xml";
```

---

I was planning to insert releases in upstream project and then remove the need to append it there

---

Maintaining the `<releases>` section upstream would be ideal, yes.